app-runner/Private/DeviceProviders/SauceLabsProvider.ps1

```diff
@@ -0,0 +1,608 @@
+            return $this.Connect()
+        }
+
+        Write-Debug "$($this.Platform): Connecting with explicit device name: $target"
+
+        # Store the device name for session creation
+        $this.DeviceName = $target
+
+        # Note: App upload and session creation happen in InstallApp
+        # because we need the App path before creating an Appium session
+
+        return @{
+            Provider    = $this
+            Platform    = $this.Platform
+            ConnectedAt = Get-Date
+            Identifier  = $target
+            IsConnected = $true
+            StatusData  = @{
+                DeviceName = $target
+                Region     = $this.Region
+                Username   = $this.Username
+            }
+        }
+    }
+
+    [void] Disconnect() {
+        Write-Debug "$($this.Platform): Disconnecting"
+
+        if ($this.SessionId) {
+            Write-Host "Ending SauceLabs session..." -ForegroundColor Yellow
+            try {
+                $sessionUri = "https://ondemand.$($this.Region).saucelabs.com/wd/hub/session/$($this.SessionId)"
+                $this.InvokeSauceLabsApi('DELETE', $sessionUri, $null, $false, $null)
+                Write-Host "Session ended successfully" -ForegroundColor Green
+            }
+            catch {
+                Write-Warning "Failed to end session: $_"
+            }
+            $this.SessionId = $null
+        }
+
+        $this.StorageId = $null
+        $this.DeviceName = $null
+        $this.CurrentPackageName = $null
+    }
+
+    [bool] TestConnection() {
+        Write-Debug "$($this.Platform): Testing connection"
+
+        # Check if we have valid credentials and session
+        if ($this.SessionId) {
+            try {
+                $baseUri = "https://ondemand.$($this.Region).saucelabs.com/wd/hub/session/$($this.SessionId)"
+                $response = $this.InvokeSauceLabsApi('GET', $baseUri, $null, $false, $null)
+                return $null -ne $response
+            }
+            catch {
+                return $false
+            }
+        }
+
+        return $null -ne $this.Username -and $null -ne $this.AccessKey
+    }
+
+    [hashtable] InstallApp([string]$PackagePath) {
+        Write-Debug "$($this.Platform): Installing App: $PackagePath"
+
+        # Validate App file
+        if (-not (Test-Path $PackagePath)) {
+            throw "App file not found: $PackagePath"
+        }
+
+        $extension = [System.IO.Path]::GetExtension($PackagePath).ToLower()
+        if ($this.MobilePlatform -eq 'Android' -and $extension -ne '.apk') {
+            throw "Package must be an .apk file for Android. Got: $PackagePath"
+        }
+        if ($this.MobilePlatform -eq 'iOS' -and $extension -ne '.ipa') {
+            throw "Package must be an .ipa file for iOS. Got: $PackagePath"
+        }
+
+        Write-Debug "Inferred platform: $($this.MobilePlatform)"
```

Comment:
It's not actually inferred - it's specified explicitly by calling the appropriate Connect-Device
```suggestion
```